client/landing/stepper/declarative-flow/internals/steps-repository/woo-confirm/index.tsx

```diff
@@ -64,49 +52,44 @@ const WooConfirm: Step = function WooCommerceConfirm( { navigation } ) {
 		requestLatestAtomicTransfer( siteId );
 	}, [ requestLatestAtomicTransfer, siteId ] );
 
-	const eligibilityHolds = useSelect(
-		( select ) =>
-			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-			// @ts-ignore Until createRegistrySelector is typed correctly
-			select( AUTOMATED_ELIGIBILITY_STORE ).getEligibilityHolds( siteId ),
-		[ siteId ]
-	);
-	const eligibilityWarnings = useSelect(
-		( select ) =>
-			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-			// @ts-ignore Until createRegistrySelector is typed correctly
-			select( AUTOMATED_ELIGIBILITY_STORE ).getEligibilityWarnings( siteId ),
-		[ siteId ]
-	);
-	const wpcomSubdomainWarning = useSelect(
-		( select ) =>
-			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-			// @ts-ignore Until createRegistrySelector is typed correctly
-			select( AUTOMATED_ELIGIBILITY_STORE ).getWpcomSubdomainWarning( siteId ),
-		[ siteId ]
-	);
-	const warnings = useSelect(
-		( select ) =>
-			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-			// @ts-ignore Until createRegistrySelector is typed correctly
-			select( AUTOMATED_ELIGIBILITY_STORE ).getNonSubdomainWarnings( siteId ),
-		[ siteId ]
-	);
-	const latestAtomicTransfer = useSelect(
-		( select ) => ( select( SITE_STORE ) as SiteSelect ).getSiteLatestAtomicTransfer( siteId || 0 ),
-		[ siteId ]
-	);
-	const latestAtomicTransferError = useSelect(
-		( select ) =>
-			( select( SITE_STORE ) as SiteSelect ).getSiteLatestAtomicTransferError( siteId || 0 ),
-		[ siteId ]
-	);
-	const productsList = useSelect(
-		( select ) => ( select( PRODUCTS_LIST_STORE ) as ProductsListSelect ).getProductsList(),
-		[]
-	);
-	const requiresUpgrade = useSelect(
-		( select ) => ( select( SITE_STORE ) as SiteSelect ).requiresUpgrade( siteId ),
+	const {
+		eligibilityHolds,
+		eligibilityWarnings,
+		wpcomSubdomainWarning,
+		warnings,
+		latestAtomicTransfer,
+		latestAtomicTransferError,
+		productsList,
+		requiresUpgrade,
+		isAtomicSite,
+		stepProgress,
+	} = useSelect(
+		( select ) => {
+			const {
+				getEligibilityHolds,
+				getEligibilityWarnings,
+				getWpcomSubdomainWarning,
+				getNonSubdomainWarnings,
+			} = select( AutomatedTransferEligibility.store );
+			const {
+				getSiteLatestAtomicTransfer,
+				getSiteLatestAtomicTransferError,
+				requiresUpgrade,
+				isSiteAtomic,
+			} = select( SITE_STORE ) as SiteSelect;
+			return {
+				eligibilityHolds: getEligibilityHolds( siteId ),
+				eligibilityWarnings: getEligibilityWarnings( siteId ),
+				wpcomSubdomainWarning: getWpcomSubdomainWarning( siteId ),
+				warnings: getNonSubdomainWarnings( siteId ),
```

Comment:
IMO the only selector that really makes sense is `getAutomatedTransferEligibility`. All others are just helper functions that extract information from the eligibility object. They could be exported from the `AutomatedTransferEligibility` module alongside the `store` object.

A very straightforward followup to this PR would be replacing the store with one simple React Query call 🙂

Reply:
> A very straightforward followup to this PR would be replacing the store with one simple React Query call 🙂

💯 I strongly agree with this. It applies to most of those stores that are actually a client-side representation of server-side data with little to no interactivity involved.